File: Proyecto/Escuela_circo/src/main/java/IFAEscuela_circo/Escuela_circo/Modelos/ProfesorCurso.java
package IFAEscuela_circo.Escuela_circo.Modelos;

import jakarta.persistence.*;
import org.hibernate.annotations.ColumnDefault;

@Entity
@Table(name = "profesor_curso")
public class ProfesorCurso {
    @Id
    @Column(name = "id", nullable = false)
    private Integer id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "profesor", nullable = false)
    private Curso profesor;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "curso", nullable = false)
    private Profesor curso;

    @ColumnDefault("0")
    @Column(name = "coordinador", nullable = false)
    private Byte coordinador;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public Curso getProfesor() {
        return profesor;
    }

    public void setProfesor(Curso profesor) {
        this.profesor = profesor;
    }

    public Profesor getCurso() {
        return curso;
    }

    public void setCurso(Profesor curso) {
        this.curso = curso;
    }

    public Byte getCoordinador() {
        return coordinador;
    }

    public void setCoordinador(Byte coordinador) {
        this.coordinador = coordinador;
    }

}

File: Proyecto/AcademyGestor/AcademyGestor/Modelos/Profesor_Curso.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AcademyGestor.Modelos
{
    class Profesor_Curso
    {
        private int id;
        private Profesor profesor;
        private Curso curso;
        private bool coordinador;

        public int Id { get => id; set => id = value; }
        public Profesor Profesor { get => profesor; set => profesor = value; }
        public Curso Curso { get => curso; set => curso = value; }
        public bool Coordinador { get => coordinador; set => coordinador = value; }
    }
}
